core/game.py

```diff
@@ -17,13 +18,25 @@ class Game:
     def __init__(self, player_names: list):
         self.players = self.__generate_players(player_names)
         self.round = 1
-        self.cards_remaining = self.__total_trait_cards
+        self.cards_remaining = [TraitCard(name=f'Trait{i}') for i in range(177)]  # Todo: Replace with real traits.
         self.phase = Phase.DEAL
 
     def __generate_players(self, player_names: list) -> list:
         if len(set(player_names)) != len(player_names):
             raise ValueError(f'Duplicate player names found: {player_names}')
         elif len(player_names) > self.__max_players:
-            raise ValueError(f'Too many players: {len(player_names)}. The maximum is: {self.__max_players}')
+            raise ValueError(
+                f'Too many players: {len(player_names)}. '
+                f'The maximum is: {self.__max_players}')
         else:
-            return [Player(player_name) for player_name in player_names]
\ No newline at end of file
+            return [Player(player_name) for player_name in player_names]
+
+    def deal_cards(self):
+        for player in self.players:
```

Comment:
Should we add the check on number of available cards? Otherwise we start doing work when it might actually be the end of the game?

Reply:
Yes.

Reply:
Should we depend on : has_enough_cards_to_deal_sufficient_cards_to_each_player? And do nothing if insufficient? Perhaps returning True/False as appropriate?

Reply:
Yes